Refactor/sync levels v2

adamscarberry opened the pull request.

adamscarberry committed:
WIP

adamscarberry committed:
WIP

adamscarberry committed:
this should work now, needs further testing

jeffsuperglide committed:
Sync CWMS Levels Refactor

Sync CWMS Levels DAG has both creating and updating in the same DAG.
A create levels list is created when not in Water and updated after
the create task finishes.

Full testing needed but looks like the DAG will work. . There seems
to be an issue with `v_locations`.  There are no entries for provider =
lrb.  Locations associated with LRB don't seem to be represented.

jeffsuperglide committed:
WIP: CWMS Levels

Refactor sync CWMS levels

adamscarberry commented:
@jeffsuperglide 

![image](https://user-images.githubusercontent.com/46494851/204393360-beb4c81b-063c-4cd9-81e1-902d82ac19e3.png)

Wondering about `water_hook.request(endpoint=payload)`.  Endpoint and payload seem like two different things.

https://github.com/USACE/airflow-config/blob/a8b5e1401b4318bc9825559ff89d270de3c8833f/dags/water/a2w_sync_cwms_levels.py#L128-L131

jeffsuperglide commented:
That is where I left off before 🦃 and they are supposed to be separate args for `endpoint` and `method`.

jeffsuperglide committed:
WIP: Test LRB worked for create and update measurements

jeffsuperglide committed:
Update sync locations and levels

Test MSCs LRB, NWD, and NWS Sync Levels

jeffsuperglide committed:
Remove comments no longer needed

jeffsuperglide committed:
Set for loops creating TaskGroup with all MSCs

jeffsuperglide committed:
DUMMY trigger rule

adamscarberry committed:
update timeseries measurements, needs further testing

adamscarberry committed:
change schedule_interval to schedule

adamscarberry committed:
rename dag and file

adamscarberry committed:
upgrade local env to airflow 2.4.3

adamscarberry committed:
cleanup dags and radar functions

jeffsuperglide committed:
Locations, Levels and Timeseries Sync

Updates to DAGs include:

- a2w_sync_cwms_locaitons
- a2w_sync_cwms_levels
- a2w_sync_cwms_timeseries (a.k.a. timeseries_paths)
- warning messages for already registered tasks solved

jeffsuperglide committed:
Rename timeseries_paths to cwms_timeseries

jeffsuperglide committed:
Merge branch 'refactor/sync-levels-v2' of https://github.com/USACE/airflow-config into refactor/sync-levels-v2

adamscarberry committed:
update dag to work, but only adds new sites, no updates

adamscarberry committed:
WIP - sync usgs timeseries

adamscarberry committed:
adjustments to usgs timeseries dag

adamscarberry committed:
delete old usgs dag
